### PR TITLE
VEN-108 Auto-select available on and shipping category in Product form

### PR DESCRIPTION
**Pull request by aleksandarpetrushev**

### CODE REVIEW

File: app/views/spree/admin/products/new.html.erb [hunk 1]
```diff
@@ -66,7 +66,7 @@
       <div data-hook="new_product_shipping_category" class="col-12 col-md-4">
         <%= f.field_container :shipping_category do %>
           <%= f.label :shipping_category_id, Spree.t(:shipping_categories) %><span class="required">*</span>
-          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
+          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { selected: @shipping_categories.first.id, include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
```
Comment (damianlegawiec) [inline app/views/spree/admin/products/new.html.erb:69]:
```suggestion
          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { selected: @shipping_categories&.first&.id, include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
```

Comment (damianlegawiec) [reply app/views/spree/admin/products/new.html.erb:69]: @aleksandarpetrushev I would rather move setting these default values to controller

**Review by damianlegawiec — COMMENTED** [review]
@aleksandarpetrushev let's move setting these default values to controller please

File: app/views/spree/admin/products/new.html.erb [hunk 2]
```diff
@@ -66,7 +66,7 @@
       <div data-hook="new_product_shipping_category" class="col-12 col-md-4">
         <%= f.field_container :shipping_category do %>
           <%= f.label :shipping_category_id, Spree.t(:shipping_categories) %><span class="required">*</span>
-          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
+          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { selected: @product.shipping_category&.id, include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
```
Comment (damianlegawiec) [inline app/views/spree/admin/products/new.html.erb:69]:
```suggestion
          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { selected: @product.shipping_category_id }, { class: 'select2 w-100' }) %>
```

File: app/controllers/spree/admin/products_controller.rb [hunk 3]
```diff
@@ -11,6 +11,12 @@ class ProductsController < ResourceController
       update.before :update_before
       helper_method :clone_object_url
 
+      def new
+        @product.available_on ||= Time.current
```
Comment (damianlegawiec) [inline app/controllers/spree/admin/products_controller.rb:15]: @aleksandarpetrushev let's move that into a before action and not overwrite the new method - it should be cleaner :)

File: app/views/spree/admin/products/new.html.erb [hunk 4]
```diff
@@ -66,7 +66,7 @@
       <div data-hook="new_product_shipping_category" class="col-12 col-md-4">
         <%= f.field_container :shipping_category do %>
           <%= f.label :shipping_category_id, Spree.t(:shipping_categories) %><span class="required">*</span>
-          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
+          <%= f.collection_select(:shipping_category_id, @shipping_categories, :id, :name, { selected: @product.shipping_category_id, include_blank: Spree.t('match_choices.none') }, { class: 'select2 w-100' }) %>
```
Comment (damianlegawiec) [inline app/views/spree/admin/products/new.html.erb:69]: @aleksandarpetrushev let's remove the include blank option, we always want the product to be associated to a shipping category, without shipping category product cannot be shipped :)

**Comment by aleksandarpetrushev** [comment]
@damianlegawiec these 3 tests (order_details_spec > as Fakedispatch) are failing, seems like with this error:
Selenium::WebDriver::Error::UnexpectedAlertOpenError:
  unexpected alert open: {Alert text : Doorkeeper::Errors::DoorkeeperError}
 
Can not get them to fail locally and keep failing on CI. Any idea why it may be?

**Comment by damianlegawiec** [comment]
> @damianlegawiec these 3 tests (order_details_spec > as Fakedispatch) are failing, seems like with this error: Selenium::WebDriver::Error::UnexpectedAlertOpenError: unexpected alert open: {Alert text : Doorkeeper::Errors::DoorkeeperError}
> 
> Can not get them to fail locally and keep failing on CI. Any idea why it may be?

you need to rebase with main branch

File: spec/features/admin/orders/order_details_spec.rb [hunk 5]
```diff
@@ -628,7 +628,7 @@
       can [:admin, :manage, :read, :ship], Spree::Shipment
     end
 
-    let(:admin_app) { Spree::OauthApplication.create(name: 'Admin Panel', scopes: 'admin', redirect_uri: current_url) }
+    let(:admin_app) { Spree::OauthApplication.create(name: 'Admin Panel', scopes: 'admin', redirect_uri: 'urn:ietf:wg:oauth:2.0:oob') }
```
Comment (damianlegawiec) [inline spec/features/admin/orders/order_details_spec.rb:631]:
```suggestion
    let(:admin_app) { Spree::OauthApplication.create(name: 'Admin Panel', scopes: 'admin', redirect_uri: '') }
```

will also work

**Comment by damianlegawiec** [comment]
@aleksandarpetrushev this PR is very messy - please create a new one

**Pull request closed without merging**